pandas/tests/series/methods/test_isin.py

```diff
@@ -220,3 +220,16 @@ def test_isin_complex_numbers(array, expected):
     # GH 17927
     result = Series(array).isin([1j, 1 + 1j, 1 + 2j])
     tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "data,is_in",
+    [([1, [2]], [1]), (["simple str", [{"values": 3}]], ["simple str"])],
+)
+def test_isin_filtering_with_mixed_object_types(data, is_in):
+
```

Comment:
```suggestion
    # GH 20883
```

Can add the original issue number here as reference

Reply:
Thanks! Resolved